e2e/cypress/integration/integrations/builtin_commands/common_commands_spec.js

```diff
@@ -288,6 +288,34 @@ describe('I18456 Built-in slash commands: common', () => {
         cy.uiWaitUntilMessagePostedIncludes(`You were added to the channel by @${user1.username}`);
     });
 
+    it('MM-T659 /invite - other channel', () => {
+        const userToInvite = userGroup[3];
+
+        loginAndVisitDefaultChannel(user1, `${team1.name}/channels/${testChannel.name}`);
+        cy.postMessage('Hello World!');
```

Comment:
I need to post a message to the channel before having the permission to add a new user to it.

Reply:
I think this is because the page hasn't fully loaded yet. I replaced it with the one below and seem to pass consistently for me.

```
cy.get('#postListContent', {timeout: TIMEOUTS.HALF_MIN}).should('be.visible');
```

Reply:
I can't get the test to pass on my end even with the timeout. Would there be a REST API call that we should watch out for instead of waiting half a minute?

I've found out that I don't have to post a message to the channel, but I need to visit the channel prior before having the permission to /invite someone to it.

Reply:
I re-read the test case and I see what you're saying now. `user1` needs to be part of the channel it's trying to invite someone else to join (either visiting that channel like before or calling the `apiAddUserToChannel`),

1. Add user1 to testChannel
2. User1 visits team town-square and post the invite slash command for user2 to join testChannel
3. User2 visits team town-square and verifies it sees testChannel is added and unread

Maybe something similar to this,

```
    it('MM-T659 /invite - other channel', () => {
        const userToInvite = userGroup[3];

        cy.apiAddUserToChannel(testChannel.id, user1.id);
        loginAndVisitDefaultChannel(user1, `${team1.name}/channels/town-square`);
        cy.get('#postListContent', {timeout: TIMEOUTS.HALF_MIN}).should('be.visible');

        // # Post `/invite @username ~channel` where channelname is a channel you have permission to add members to but not the current channel, and username is a user not in that other channel
        cy.postMessage(`/invite @${userToInvite.username} ~${testChannel.name}`);

        // * User who added them sees system message "username added to channelname channel."
        cy.uiWaitUntilMessagePostedIncludes(`${userToInvite.username} added to ${testChannel.name} channel.`);

        loginAndVisitDefaultChannel(userToInvite, `${team1.name}/channels/town-square`);
        cy.get('#postListContent', {timeout: TIMEOUTS.HALF_MIN}).should('be.visible');

        // * Added user sees channel added to LHS, mention badge.
        cy.get('#sidebarChannelContainer').
            find(`[href*="${team1.name}/channels/${testChannel.name}"]`).
            within(() => {
                cy.get('#unreadMentions').should('be.visible');
                cy.findByText(`${testChannel.display_name}`).click();
            });

        // * Added user sees system message "username added to the channel by username."
        cy.uiWaitUntilMessagePostedIncludes(`You were added to the channel by @${user1.username}`);
    });
```

Reply:
Nice! I should've thought of using an API call to add a user to the channel.